src/Compilers/Core/Portable/Symbols/ManagedKind.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.CodeAnalysis
+{
+    /// <summary>
+    /// PROTOTYPE
+    /// </summary>
+    [Flags]
+    internal enum ManagedKind : byte
+    {
+        Unknown = 0,
+        Unmanaged = 1,
+        UnmanagedWithGenerics = 2,
```

Comment:
I don't understand why we need this value... where is it used?

Reply:
https://github.com/dotnet/roslyn/pull/31602/files#diff-a921499c9550563c162f3e1cbdff548bR1151

Reply:
Ah, I see -- and you need to see not only if the type itself is generic, you also need to see if any of the fields, transitively, are generic?

Reply:
Exactly. This is also why I added the `ref bool hasGenerics` parameter to the method that does that traversal.

Reply:
https://github.com/dotnet/roslyn/pull/31602/files/87d19f770a84f30034d4c519b2a3b23d9129fd88#diff-723bd6bd1611905114d8dcbc9431f1a3L126